components/frontend/src/subject/SubjectTableRow.js

```diff
@@ -25,28 +26,97 @@ import {
     stringsPropType,
 } from '../sharedPropTypes';
 
-function MeasurementCells({ dates, metric, metric_uuid, measurements }) {
+function didValueIncrease(dateOrderAscending, metricValue, previousValue) {
+    return (dateOrderAscending && metricValue > previousValue) || (!dateOrderAscending && metricValue < previousValue)
+}
+didValueIncrease.propTypes = {
+    dateOrderAscending: PropTypes.bool,
+    metricValue: measurementValueType,
+    previousValue: measurementValueType,
+}
+
+function didValueImprove(didValueIncrease, direction) {
+    return (didValueIncrease && direction === ">") || (!didValueIncrease && direction === "<")
+}
+didValueImprove.propTypes = {
+    didValueIncrease: PropTypes.bool,
+    direction: PropTypes.oneOf(["<", ">"]),
+}
+
+function DeltaCell({ dateOrderAscending, index, metric, metricValue, previousValue, status }) {
+    const dataModel = useContext(DataModel);
+    let label = null;
+    if (index > 0 && previousValue !== "?" && metricValue !== "?" && previousValue !== metricValue) {
+        const direction = getMetricDirection(metric, dataModel)
+        const increased = didValueIncrease(dateOrderAscending, metricValue, previousValue)
+        const improved = didValueImprove(increased, direction)
+        const evaluateTarget = metric.evaluate_targets ?? true
+        let alt = "The measurement value changed";
+        let color = "blue";
+         if (evaluateTarget && improved) {
+            alt = "The measurement value improved";
+            color = "green"
+        }
+        if (evaluateTarget && !improved) {
+            alt = "The measurement value worsened";
```

Comment:
Is it okay to not address the case where the value stayed the same? Do we skip that?